### PR TITLE
Support for mixed celltypes

### PR DESCRIPTION
**Pull request by JimBrouzoulis**
This PR is to discuss:

* an example implementation of having mixed celltypes 
* other changes required in the code

Example script using the MixedGrid
https://gist.github.com/JimBrouzoulis/98e5b8e50895ea322da354eb0ab197ba

### CODE REVIEW

File: src/Dofs/MixedDofHandler.jl [hunk 1]
```diff
@@ -0,0 +1,254 @@
+
+mutable struct FieldHandler
+    field_names::Vector{Symbol}
```
Comment (lijas) [inline src/Dofs/MixedDofHandler.jl]: Maybe add a `struct Field` with`field_name`, `field_dims` and `and field_interpolations`

File: src/Dofs/MixedDofHandler.jl [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    field_dims::Vector{Int}
+    field_interpolations::Vector{JuAFEM.Interpolation}
+    quad_rules::Vector{JuAFEM.QuadratureRule}
```
Comment (lijas) [inline src/Dofs/MixedDofHandler.jl]: I dont think QuadratureRule should be stored here. User should handle this on their own?

File: src/Dofs/MixedDofHandler.jl (hunk 1 above)
Comment (JimBrouzoulis) [reply src/Dofs/MixedDofHandler.jl]: I have done so now. The FieldHandler stores a number of Fields.

File: src/Dofs/MixedDofHandler.jl (hunk 2 above)
Comment (JimBrouzoulis) [reply src/Dofs/MixedDofHandler.jl]: Yes, I agree. I don't know why I added it there.

**Comment by fredrikekre** [comment]
:tada: 

I think we can probably combine the two types in the future, but let's try it out separately first.

**Comment by KristofferC** [comment]
Woot!

**Comment by KristofferC** [comment]
Need an example now :)

**Comment by JimBrouzoulis** [comment]
> Need an example now :)

A small test that can be reworked into an example. This can then be used to benchmark and find bottlenecks in the current implementation.
https://github.com/KristofferC/JuAFEM.jl/blob/07f6d065865cacc891909261b2bd243ae80b1ef2/test/test_mixeddofhandler.jl#L295

**Pull request merged**